tests/org.eclipse.n4js.xpect.tests/model/validation/dataflow/NullUndefined_May_OptionalChaining.n4js.xt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+
+/*	XPECT_SETUP org.eclipse.n4js.xpect.tests.N4JSXpectTest 
+	IssueConfiguration {
+		IssueCode "DFG_NULL_DEREFERENCE" {enabled=true}
+	}
+	END_SETUP
+ */
+
+class C {
+	field: C;
+}
+
+
+function f1() {
+	let c : C;
+	// XPECT warnings --> "Variable c is undefined" at "c"
+	let x = c.field;
+}
+
+
+function f2() {
+	let c : C;
+	if (0) c = new C();
+	// XPECT warnings --> "Variable c may be undefined" at "c"
+	let x = c.field;
+}
+
+
+function f3() {
+	let c : C;
+	if (0) c = new C();
+	// XPECT nowarnings --> "Variable c is undefined" at "c"
+	let x = c?.field;
+}
+
+
+function f4() {
+	let c : C;
+	if (0) c = new C();
+	let x = c?.field;
+	// XPECT warnings --> "Variable x may be undefined" at "x"
+	x.toString();
+}
+
+
+function f5() {
+	let c : C;
+	let x = c?.field;
+	// XPECT warnings --> "Variable x may be undefined" at "x"
+	x.toString();
```

Comment:
I think in this case it would make sense to warn with `Variable x is(!) undefined`.